osu.Game/Rulesets/UI/DrawableRuleset.cs

```diff
@@ -214,11 +214,14 @@ private void loadObjects(CancellationToken? cancellationToken)
         {
             foreach (TObject h in Beatmap.HitObjects)
             {
-                cancellationToken?.ThrowIfCancellationRequested();
+                if (cancellationToken?.IsCancellationRequested == true)
+                    return;
+
                 addHitObject(h);
             }
 
-            cancellationToken?.ThrowIfCancellationRequested();
+            if (cancellationToken?.IsCancellationRequested == true)
```

Comment:
no need for `== true` :')

Reply:
have you tried this? i think you'll find there is.

Reply:
I only looked at the code, my bad if that's correct

Reply:
2 points:
- For this expression that results in `bool?`, `== true` is meaningful.
- But the usage of `CancellationToken?` is unnecessary.  `CancellationToken.None` and `default(CancellationToken)` both mean "will never be canceled", and widely used in BCL.

I'll send a PR for this.

Reply:
You're welcome to look into that, but I believe it is nullable for a reason (to do with DI).